### ISSUE TITLE
Publish types for setProperty

### ISSUE THREAD

**Issue by sqs** [microsoft/node-jsonc-parser#20]
We use the [`setProperty`](https://sourcegraph.com/search?q=repo:%5Egithub%5C.com/sourcegraph/sourcegraph%24+%5B%5E.%5DsetProperty%5C%28+lang:typescript) function by itself. Right now, we use a fork of node-jsonc-parser because the main package doesn't publish types for non-main modules. Would you accept a PR that publishes types for all of the `.js` files in the npm package?

We use `setProperty` for the quick action buttons in the JSON settings editors on Sourcegraph, in case anyone's curious about the specific use case: 
![image](https://user-images.githubusercontent.com/1976/57586647-989b7580-74ad-11e9-9225-927f06919084.png)

**Comment by aeschli** [comment]
No problem for me to make the rewriter APIs public. It should be done by adding it to the `main.ts` class. The files in `impl` are considered implementation details so no d.ts for that.

**Comment by aeschli** [comment]
I'm happy to take a PR if you want to give it a try.

**Comment by yowayb** [comment]
`setProperty()` is exported as `modify()` with the exact same function signature, plus types.  Please correct me if I'm wrong, but I think this means we should use `modify()` instead and discard this issue.

**Comment by aeschli** [comment]
Yes, `modify` is the API that is exposed, and it what should be used. `setProperty` is an internal method.
